src/streamlink/plugins/pandalive.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import re
+
+from streamlink.plugin import Plugin, PluginError, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.hls import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(re.compile(
+    r"https?://(?:www\.)?pandalive\.co\.kr/"
+))
+class Pandalive(Plugin):
+    API_URL = "https://api.pandalive.co.kr/v1/live/play"
+
+    _room_id_re = re.compile(r"roomid\s*=\s*String\.fromCharCode\((.*)\)")
+
+    def _get_media_code(self):
+        root = self.session.http.get(self.url, schema=validate.Schema(
+            validate.parse_html(),
+        ))
+
+        try:
+            schema = validate.Schema(
+                validate.xml_xpath_string(".//script[contains(text(), 'roomid')]/text()"),
+                str,
+                validate.transform(self._room_id_re.search),
+                validate.get(1),
+                str,
+                validate.transform(lambda s: "".join(map(lambda c: chr(int(c)), s.split(",")))),
+            )
+            return schema.validate(root)
+        except PluginError:
+            pass
+
+        if not root.text:
+            log.error("The broadcast has ended")
```

Comment:
Can this even be true? That would mean that the entire page (`self.url`) does have no content.
On invalid URLs (matching non-stream URLs), "The broadcast has ended" is also wrong.

```
$ streamlink 'https://www.pandalive.co.kr/live/play/'
[cli][info] Found matching plugin pandalive for URL https://www.pandalive.co.kr/live/play/
[plugins.pandalive][error] The broadcast has ended
error: No playable streams found on this URL: https://www.pandalive.co.kr/live/play/
```

Reply:
Hmm!  Maybe I should just drop that detection and return without any message.  It seems like the only way to keep it would be to hard code the Korean language string returned into the plugin, which I'm guessing isn't ideal.

See the source of [this](https://www.pandalive.co.kr/live/play/?HFX9o9ZQl32zAu2GFv1uthl=16838722) past stream as an example and compare to the source of your link.

What do you think?  I'm guessing probably dropping is the way forward?

Reply:
Yeah, you could check for `방송이 종료되었습니다.` and only then log the message, but since it's an optional message, I don't think it's too important. If a stream is offline or unavailable for whatever reason, then all that Streamlink has to do is return no streams, so these kinds of log messages are rather pointless and only "useful" for the CLI anyway.